mono/utils/mono-dl-posix.c

```diff
@@ -132,12 +132,18 @@ mono_dl_lookup_symbol (MonoDl *module, const char *name)
 int
 mono_dl_convert_flags (int flags)
 {
-	int lflags = flags & MONO_DL_LOCAL? 0: RTLD_GLOBAL;
+	int lflags = 0;
+
+	if (flags & MONO_DL_LOCAL)
+		flags |= RTLD_LOCAL;
+	else if (flags & MONO_DL_GLOBAL)
```

Comment:
What is the expected behavior for specifying local and global?

Reply:
I figured we should make it default to local, but I actually have no idea what dlopen will do, that's a good question. Maybe I should just pass on both...

Reply:
I'm guessing one of them is zero so dlopen cannot tell.
But I don't know.
Maybe error, or comment.
You just don't want accidental semantics, that you are then stuck with.

Reply:
Indeed local is zero, with glibc.
```
C:\s>type 1.c
#include <dlfcn.h>

int a = RTLD_LOCAL;
int b = RTLD_GLOBAL;

C:\s>wsl gcc -E 1.c
int a =
       0
                 ;
int b =
       0x00100
```

Reply:
Even if it is, you still have the lazy/now split, so both cannot be zero. They have to take a stand on an issue like this somewhere, though maybe the behavior is just unspecified.

I'm inclined to just comment that local takes precedence and that it was a conscious decision, but I don't feel strongly on this.